pkg/resourceinterpreter/configurableinterpreter/configmanager/interpreters.go

```diff
@@ -0,0 +1,124 @@
+}
+
+var _ Interpreter = (*interpreterManagers)(nil)
+
+func (i interpreterManagers) LoadConfig(customizations configv1alpha1.CustomizationRules) error {
+	for _, interpreter := range i.interpreters {
+		err := interpreter.LoadConfig(customizations)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (i interpreterManagers) IsEnabled(operation configv1alpha1.InterpreterOperation) bool {
+	for _, interpreter := range i.interpreters {
+		if interpreter.IsEnabled(operation) {
+			return true
+		}
+	}
+	return false
+}
+
+func (i interpreterManagers) Retain(desired *unstructured.Unstructured, observed *unstructured.Unstructured) (retained *unstructured.Unstructured, handled bool, err error) {
+	for _, interpreter := range i.interpreters {
+		retained, handled, err = interpreter.Retain(desired, observed)
+		if err != nil || handled {
+			return
+		}
+	}
+	return
+}
+
+func (i interpreterManagers) GetReplicas(object *unstructured.Unstructured) (replicas int32, require *workv1alpha2.ReplicaRequirements, handled bool, err error) {
+	for _, interpreter := range i.interpreters {
+		replicas, require, handled, err = interpreter.GetReplicas(object)
+		if err != nil || handled {
+			return
+		}
+	}
+	return
+}
+
+func (i interpreterManagers) ReviseReplica(object *unstructured.Unstructured, replica int64) (obj *unstructured.Unstructured, handled bool, err error) {
+	for _, interpreter := range i.interpreters {
+		obj, handled, err = interpreter.ReviseReplica(object, replica)
+		if err != nil || handled {
+			return
+		}
+	}
+	return
+}
+
+func (i interpreterManagers) AggregateStatus(object *unstructured.Unstructured, aggregatedStatusItems []workv1alpha2.AggregatedStatusItem) (obj *unstructured.Unstructured, handled bool, err error) {
+	for _, interpreter := range i.interpreters {
+		obj, handled, err = interpreter.AggregateStatus(object, aggregatedStatusItems)
+		if err != nil || handled {
+			return
+		}
+	}
+	return
+}
+
+func (i interpreterManagers) GetDependencies(object *unstructured.Unstructured) (dependencies []configv1alpha1.DependentObjectReference, handled bool, err error) {
+	for _, interpreter := range i.interpreters {
+		dependencies, handled, err = interpreter.GetDependencies(object)
+		if err != nil || handled {
+			return
+		}
+	}
+	return
+}
+
+func (i interpreterManagers) ReflectStatus(object *unstructured.Unstructured) (status *runtime.RawExtension, handled bool, err error) {
+	for _, interpreter := range i.interpreters {
+		status, handled, err = interpreter.ReflectStatus(object)
+		if err != nil || handled {
+			return
+		}
+	}
+	return
+}
+
+func (i interpreterManagers) InterpretHealth(object *unstructured.Unstructured) (healthy bool, handled bool, err error) {
+	for _, interpreter := range i.interpreters {
+		healthy, handled, err = interpreter.InterpretHealth(object)
+		if err != nil || handled {
+			return
+		}
+	}
+	return
+}
+
+func toUnstructured(v engine.Value) (*unstructured.Unstructured, error) {
+	if v.IsNil() {
```

Comment:
When the return value is nil, do we need to return an error? Because for most cases, the logic after returning nil will be wrong.

Reply:
all are fixed